src/requirements.ts

```diff
@@ -145,3 +168,37 @@ function invalidJavaHome(reject, cause: string) {
     });
   }
 }
+
+export async function installManagedJre() {
+  return vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: 'Downloading JRE' },
+    async (progress, cancelToken) => {
+      const platformInfo = await PlatformInformation.GetPlatformInformation();
```

Comment:
When an error happens in an `await`ed thing, an exception is thrown. You can add a try/catch.